ruledata.go

```diff
@@ -716,6 +688,38 @@ func SetRules(s *NetlinkConnection, content []byte) error {
 					}
 				}
 			}
+		case "file_rules":
+			vi := v.([]interface{})
+			for ruleNo := range vi {
+				rule := vi[ruleNo].(map[string]interface{})
+				path := rule["path"]
+				if path == "" {
+					log.Fatalln("Watch option needs a path!")
+				}
+				// TODO: Add key addition support
+				// key := rule["key"]
+				perms := rule["permission"]
+				log.Println("Setting File Permissions", path, perms)
+				var dd AuditRuleData
+				dd.Buf = make([]byte, 0)
+				add := AUDIT_FILTER_EXIT
+				action := AUDIT_ALWAYS
+				_audit_syscalladded = true
+
+				err := AuditSetupWatchName(&dd, path.(string))
```

Comment:
I think you should call AuditAddWatchDir here and inside that function call AuditSetupWatchName.
Since, AuditAddWatchDir is the API function which will be called by programs like Audit-go and Auditctl-go.
Also, the AuditSetupWatchName should be renamed to auditSetupWatchName.

Similary with AuditAddWatchPerm